samples/algorithms/PhaseEstimation.qs

```diff
@@ -0,0 +1,86 @@
+/// # Sample
+/// Quantum Phase Estimation
+///
+/// # Description
+/// This sample demonstrates how to use Quantum Phase Estimation (QPE) algorithm
+/// to estimate the eigenvalue of a specific unitary operation.
+/// QPE algorithm is performed by calling `ApplyQPE` library operation.
+/// Run this sample and check the histogram of results.
+
+import Std.Math.*;
+
+/// # Summary
+/// Estimate the the eigenvalue of a specific unitary U for its eigenvector |010⟩
+/// using `ApplyQPE`. The result is returned as a complex polar value.
+operation Main() : ComplexPolar {
+    // Allocate qubits to be used in phase estimation
+    use state = Qubit[3];
+
+    // Prepare U eigenvector |010⟩
+    X(state[1]);
+
+    // Estimate eigenvalue of U corresponding eigenvector |010⟩
+    let eigenvalue = EstimateEigenvalue(U, state, 6);
+
+    // Reset the state qubits
+    ResetAll(state);
+
+    // The eigenvalue represented as ComplexPolar should be (1.0, π/3.0)
+    // or approximately (1.0, 1.04719755).
+    eigenvalue
+}
+
+/// # Summary
+/// Estimate the eigenvalue of a unitary operation `U` with certain `precision`
+/// for a given eigenvector using Quantum Phase Estimation (QPE) algorithm.
+operation EstimateEigenvalue(
+    U : Qubit[] => Unit is Adj + Ctl,
+    eigenstate : Qubit[],
+    precision : Int
+) : ComplexPolar {
+    // Allocate qubits to be used in phase estimation
+    use phase = Qubit[precision];
```

Comment:
It would be useful to add an example for iterative phase estimation that only uses a single helper qubit. It's more commonly used than this version of phase estimation that applies the QFT all at once on multiple helper qubits.

Reply:
Yes! More phase estimation approaches and samples are in the backlog. However, I added a standard one to Canon namespace as it is unlikely to change. Other approaches will be in a separate library later.